test/validations/bigint.spec.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * @adonisjs/validator
+ *
+ * (c) Harminder Virk <virk@adonisjs.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { test } from '@japa/runner'
+import { rules } from '../../src/Rules'
+import { validate } from '../fixtures/rules/index'
+import { MessagesBag } from '../../src/MessagesBag'
+import { ApiErrorReporter } from '../../src/ErrorReporter'
+import { bigint } from '../../src/Validations/primitives/bigint'
+
+function compile() {
+  return bigint.compile('literal', 'bigint', rules['bigint']().options, {})
+}
+
+test.group('BigInt', () => {
+  validate(bigint, test, 'helloworld', 10n, compile())
+
+  test('report error when value is near Infinity', ({ assert }) => {
+    const reporter = new ApiErrorReporter(new MessagesBag({}), false)
+    bigint.validate(
+      '-3177777777777777777777777777777777777777777777777777777777777777777777777770000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999991111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111',
+      compile().compiledOptions,
+      {
+        errorReporter: reporter,
+        field: 'age',
+        pointer: 'age',
+        tip: {},
+        root: {},
+        refs: {},
+        mutate: () => {},
+      }
+    )
+
+    assert.deepEqual(reporter.toJSON(), {
+      errors: [
+        {
+          field: 'age',
+          rule: 'bigint',
+          message: 'bigint validation failed',
+        },
+      ],
+    })
+  })
+
+  test('report error when value is not a valid bigint', ({ assert }) => {
+    const reporter = new ApiErrorReporter(new MessagesBag({}), false)
+    bigint.validate(null, compile().compiledOptions, {
+      errorReporter: reporter,
+      field: 'age',
+      pointer: 'age',
+      tip: {},
+      root: {},
+      refs: {},
+      mutate: () => {},
+    })
+
+    assert.deepEqual(reporter.toJSON(), {
+      errors: [
+        {
+          field: 'age',
+          rule: 'bigint',
+          message: 'bigint validation failed',
+        },
+      ],
+    })
+  })
+
+  test('cast bigint like string to a valid bigint', ({ assert }) => {
+    const reporter = new ApiErrorReporter(new MessagesBag({}), false)
+    let value: any = '21'
```

Comment:
Should we also accept a string value ending with `n`?

Reply:
`BigInt()` function doesn't allow a string value ending by `n`

```ts
BigInt('12')
// 12n

BigInt('12n')
// error
```
We can do it but it's not the default behavior of `BigInt`, dunno if it's a good idea